os/fs/smartfs/smartfs_smart.c

```diff
@@ -277,9 +280,51 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
 
 		if (parentdirsector != 0xFFFF) {
 			/* We can create in the given parent directory */
+			/* First we allocate a new data sector for the file */
+			ret = smartfs_alloc_firstsector(&data_sector); //More attributes to be added
+			if (ret != OK) {
+				fdbg("\nError allocating new data sector for file\n");
+				goto errout_with_buffer;
+			}
+			fdbg("\nAllocated firstsector = %d\n", data_sector);
+
+			/* First try to find an invalid or empty entry available in one of the chained parent sectors */
+			ret = smartfs_findentry(fs, parentdirsector, &newentry);
```

Comment:
it is very similar with finddirentry..
I really don't like invalidentry,,,but it seems that no way..
because of ENTRY_VALID MARO

Reply:
I was also thinking that the name can be confusing.
Lets call it smartfs_find_availableentry or smartfs_find_unusedentry?

Also, VALID_ENTRY macro may be split up into two parts
1. IS_VALID_ENTRY - checks if flags are invalidated or not
2. A second check for if flags are in erased state